smithy-model/src/main/java/software/amazon/smithy/model/node/NumberNode.java

```diff
@@ -94,6 +95,38 @@ public Optional<NumberNode> asNumberNode() {
         return Optional.of(this);
     }
 
+    /**
+     * Returns true if the value of the number contained in the number node is zero,
+     * accounting for float, double, bigInteger, bigDecimal, and other numeric types
+     * (e.g., 0, 0.0, 0.0000000, etc).
+     *
+     * @return Returns true if set to zero.
+     */
+    public boolean isZero() {
+        // Do a cheap test based on the serialized value of the number first.
+        if (toString().equals("0")) {
+            return true;
+        } else if (value instanceof BigDecimal) {
+            return value.equals(BigDecimal.ZERO);
+        } else if (value instanceof BigInteger) {
+            return value.equals(BigInteger.ZERO);
+        } else if (value instanceof Float) {
+            // Using Math.signum to avoid epsilon issues.
```

Comment:
If there is a test case that shows the epsilon issue that would have broken without this behavior, that'd be great.